File: go/0049_group_anagrams.go
func groupAnagrams(strs []string) [][]string {
	m := make(map[[26]int][]string)
	for i := 0; i < len(strs); i++ {
		str := strs[i]

		var k [26]int
		for j := 0; j < len(str); j++ {
			k[(str[j]-'a')]++
		}

		if _, ok := m[k]; ok {
			m[k] = append(m[k], str)
		} else {
			m[k] = []string{str}
		}
	}

	var res [][]string
	for _, v := range m {
		res = append(res, v)
	}
	return res
}

File: python/0049_group_anagrams.py
from collections import defaultdict


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = defaultdict(lambda: [])
        for s in strs:
            ht = [0] * 26
            for ch in s:
                ht[ord(ch) - ord('a')] += 1
            mp[self.stringify(ht)].append(s)

        return [x for x in mp.values()]

    def stringify(self, ht):
        res = ''
        for i in range(26):
            if ht[i] > 0:
                res += chr(i + ord('a'))
                res += str(ht[i])
        return res